src/jit/codegencommon.cpp

```diff
@@ -10317,10 +10332,29 @@ void CodeGen::genFuncletProlog(BasicBlock* block)
 
     compiler->unwindBegProlog();
 
+    inst_RV(INS_push, REG_FPBASE, TYP_REF);
+    compiler->unwindPush(REG_FPBASE);
+
     // TODO Save callee-saved registers
 
     // This is the end of the OS-reported prolog for purposes of unwinding
     compiler->unwindEndProlog();
+
+    if (handlerGetsXcptnObj(block->bbCatchTyp))
+    {
+        getEmitter()->emitIns_R_R(INS_mov, EA_PTRSIZE, REG_EXCEPTION_OBJECT, REG_ARG_1);
+    }
+
+    // TODO We will need changes here when we introduce PSPSym
+    getEmitter()->emitIns_R_R(INS_mov, EA_PTRSIZE, REG_FPBASE, REG_ARG_0);
+    getEmitter()->emitIns_R_AR(INS_lea, EA_PTRSIZE, REG_FPBASE, REG_FPBASE, -8);
```

Comment:
Please add a comment here about the `-8`. I assume this is because you assume the main function prolog is:
```
pushd ebp
mov ebp,esp
```
Thus, CallerSP is exactly 8 bytes above, due to the pushed return address and pushed ebp.

Note that this is the reason we have the funclet frame layout diagram comments. Also, these constants are encapsulated in other architectures with the `CodeGen::genCaptureFuncletPrologEpilogInfo()` function and descriptive names.